### PR TITLE
Allow post method validation endpoint

### PR DESCRIPTION
**Pull request by syldlb**
## Review

* [x] Tests
* [x] Docs/comments
  * [x] *IN CASE YOU'VE CHANGED THE `formidable.yml` file*: run ``tox -e swagger-statics`` to rebuild the swagger static files **and** commit the diff.
* [x] `CHANGELOG.rst` Updated
* [ ] Delete your branch

### CODE REVIEW

**Comment by syldlb** [comment]
Done in pair with @brunobord

File: docs/swagger/formidable.yml [hunk 1]
```diff
@@ -133,7 +133,21 @@ paths:
 
   /forms/{id}/validate/:
     get:
-      summary: Validate a form
+      summary: Validate a form (GET method). GET and POST are equivalent.
```
Comment (alexdashkov) [inline docs/swagger/formidable.yml]: Maybe add some comment that GET method will be removed soon? 
``` Validate a form. GET and POST are equivalent, but GET is deprecated ```

File: formidable/views.py [hunk 2]
```diff
@@ -246,12 +246,14 @@ def get(self, request, **kwargs):
...
         form_class = self.get_form_class(formidable)
-        form = self.get_form(form_class)
+        form = self.get_form(form_class, request)
         if form.is_valid():
             return self.form_valid(form)
         else:
             return self.form_invalid(form)
 
+    post = get
+
```
Comment (alexdashkov) [inline formidable/views.py]: I would like to inverse the methods (as we will remove the GET method).
```
def post(self, request, **kwargs): 
     # all the code goes here

get = post

```

Comment (syldlb) [reply formidable/views.py]: Done, also added a deprecation warning as @brunobord suggested!

File: formidable/views.py [hunk 3]
```diff
@@ -239,19 +240,31 @@ def build(self, role):
...
             return self.form_valid(form)
         else:
             return self.form_invalid(form)
 
+    def get(self, request, **kwargs):
+        """
+        GET method is deprecated in favor of POST
+
+        """
+        warnings.simplefilter('always', category=DeprecationWarning)
```
Comment (brunobord) [inline formidable/views.py]: why adding a simple filter here? let the user's default, it should be fine.

Comment (syldlb) [reply formidable/views.py]: It was done this way on PeopleAsk, I removed it

File: CHANGELOG.rst [hunk 4]
```diff
@@ -5,6 +5,7 @@ ChangeLog
 master (unreleased)
 ===================
 
+- Allow POST method for form validation endpoint.
```
Comment (brunobord) [inline CHANGELOG.rst]: FYI, there's a ``deprecations.rst`` document in the ``docs/`` directory. I think we should do the following:

1. Add a deprecation note here, in the Changelog, to inform the reader that this version (probably 1.3.0) will be the last to support the GET verb for the validation endpoint.
2. When we'll do the release, we'll add this point to the ``deprecations.rst`` document with the correct version number.

Comment (syldlb) [reply CHANGELOG.rst]: The changelog has been updated :)

File: CHANGELOG.rst [hunk 5]
```diff
@@ -5,7 +5,7 @@ ChangeLog
 master (unreleased)
 ===================
 
-- Allow POST method for form validation endpoint.
+- Allow POST method for form validation endpoint. GET method is now deprecated, it's support will be dropped in the next version.
```
Comment (brunobord) [inline CHANGELOG.rst]: I prefer active voice: 

> **Deprecation Warning:** This version is the last to support the GET verb for the validation endpoint. We'll drop it in the next version.

**Comment by brunobord** [comment]
don't forget to squash these!

**Pull request merged**